### ISSUE TITLE
Array count cannot be less than 0

### ISSUE THREAD

**Issue by SnaiperoG3D** [FirstGearGames/FishNet#802]
**General**
Unity version: 2022.3.29f1
Fish-Networking version: 4.5.0Pro, 4.5.2Pro
Discord link:
https://discord.com/channels/424284635074134018/1297578826192130050

**Description**
When class which used in network methods (TargetRpc, etc) have arrays without initialization, log `Array count cannot be less than 0` appears, which cause unexpected behaviors, such error `Remaining bytes in SyncType reader are less than expected. Something did not serialize or deserialize properly which will likely result in a SyncType being incorrect.`. Also all network methods in same tick doesnt work cuz of it.

**Replication**
Just declare array without initialization in class which used as argument in method with ServerRpc attribute or else.
```c#
    public class PlayerItem
    {
        public Vector2[] slots = new Vector2[0]; // this is fine
        public vector2[] slots; //  this cause an issue
    }

    [ServerRpc]
    public void CmdDropItem(PlayerItem _dropPlayerItem) {}  //  any network method i believe (ServerRpc, etc...)
```

**Expected behavior**
No errors and working methods.

**Screenshots**

**Comment by FirstGearGames** [comment]
Looks like the ReadArray might be incorrect. I've not been able to test this yet, please try replacing this method inside Reader.cs

```cs
        public int ReadArray<T>(ref T[] collection)
        {
            int count = (int)ReadSignedPackedWhole();

            if (count == Writer.UNSET_COLLECTION_SIZE_VALUE)
            {
                return 0;
            }
            else if (count == 0)
            {
                if (collection == null)
                    collection = new T[0];

                return 0;
            }
            else if (count < 0)
            {
                NetworkManager.Log($"Array count cannot be less than 0.");
                //Purge renaming and return default.
                Position += Remaining;
                return default;
            }
            else
            {
                //Initialize buffer if not already done.
                if (collection == null)
                    collection = new T[count];
                else if (collection.Length < count)
                    Array.Resize(ref collection, count);

                for (int i = 0; i < count; i++)
                    collection[i] = Read<T>();

                return count;
            }
        }
```

**Comment by FirstGearGames** [comment]
Resolved in 4.5.3
